### PR TITLE
fix: limit usage of * in RBAC

### PR DESCRIPTION
**Pull request by MatousJobanek**
because * includes also escalate permissions

there needs to be a follow-up on appstudio side as well

### CODE REVIEW

**Review by alexeykazakov — APPROVED** [review]

File: config/rbac/appstudio_api_permissions_role.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# permissions to scale Camel K resources.
```
Comment (mfrancisc) [inline config/rbac/appstudio_api_permissions_role.yaml:1]: is this up to date with the actual purpose of this role?

File: controllers/nstemplateset/nstemplateset_controller.go [hunk 2]
```diff
@@ -89,9 +89,8 @@ type Reconciler struct {
 //+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=nstemplatesets/status,verbs=get;update;patch
 //+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=nstemplatesets/finalizers,verbs=update
 
-//+kubebuilder:rbac:groups="",resources=namespaces;limitranges,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups="",resources=namespaces;resourcequotas,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups=rbac.authorization.k8s.io;authorization.openshift.io,resources=rolebindings;roles;clusterroles;clusterrolebindings,verbs=*
+//+kubebuilder:rbac:groups="",resources=namespaces;limitranges;resourcequotas;pods/exec,verbs=get;list;watch;create;update;patch;delete
```
Comment (mfrancisc) [inline controllers/nstemplateset/nstemplateset_controller.go:92]: just trying to understand - why is this controller doing with `pods/exec` resources ?

File: config/rbac/appstudio_api_permissions_role.yaml (hunk 1 above)
Comment (MatousJobanek) [reply config/rbac/appstudio_api_permissions_role.yaml]: Good catch. bad copy-paste - fixed in f8fbda0

File: controllers/nstemplateset/nstemplateset_controller.go (hunk 2 above)
Comment (MatousJobanek) [reply controllers/nstemplateset/nstemplateset_controller.go:92]: this is needed to be able to apply sandbox templates https://github.com/codeready-toolchain/host-operator/blob/master/deploy/templates/nstemplatetiers/base1ns/ns_dev.yaml#L21-L37

**Review by mfrancisc — APPROVED** [review]
Thanks for addressing my comments 👍

**Comment by MatousJobanek** [comment]
unfortunately, this won't be as easy as it seems to be. It's very hard to reference to an already existing ClusterRole which is no managed by OLM and it's impossible to do that in e2e (or dev) environment where we deploy two versions of member-operator. The reason is that the OLM doesn't know which SA in which namespace the ClusterRole should be assigned to. Plus, there is also a conflict in the name of the created ClusterRoleBinding.

**Comment by MatousJobanek** [comment]
I moved the PR to a draft until I figure out how to solve the problem I mentioned in my previous comment https://github.com/codeready-toolchain/member-operator/pull/440#issuecomment-1549520309

**Pull request closed without merging**